tests/bud/with-arg/Dockerfilefromarg

```diff
@@ -4,13 +4,17 @@ ARG another_app_type_default=m
 
 FROM alpine as x
 RUN echo hello
+RUN touch hello
 
 FROM ${app_type} as m
 RUN echo world
+COPY --from=0 hello .
 
 # Do not supply this in cli, lets use default
 FROM ${another_app_type_default} as final
 RUN echo hello
+COPY --from=m hello .
 
-FROM ${another_app_type} as final
+FROM ${another_app_type} as target
 RUN echo hello
+COPY --from=final hello .
```

Comment:
Why did we need to change this file to get the test to pass?  Why did the builder not figure out that the final stage needed the second stage?